tests/h/traversal/contexts_test.py

```diff
@@ -116,6 +117,51 @@ def test_acl_shared(self,
         else:
             assert not pyramid_request.has_permission('read', res)
 
+    @pytest.mark.parametrize('groupid,userid,permitted', [
+        ('freeforall', 'jim', True),
+        ('freeforall', 'saoirse', True),
+        ('freeforall', None, False),
+        ('only-saoirse', 'jim', False),
+        ('only-saoirse', 'saoirse', True),
+        ('only-saoirse', None, False),
+        ('pals', 'jim', True),
+        ('pals', 'saoirse', True),
+        ('pals', 'francis', False),
+        ('pals', None, False),
+        ('unknown-group', 'jim', False),
+        ('unknown-group', 'saoirse', False),
+        ('unknown-group', 'francis', False),
+        ('unknown-group', None, False),
+    ])
+    def test_acl_flag_shared(self,
+                            factories,
+                            pyramid_config,
+                            pyramid_request,
+                            groupid,
+                            userid,
+                            permitted,
+                            group_service,
+                            links_service):
+        """
+        Flag permissions should echo read permissions with the exception that
+        `Security.Everyone` does not get the permission
+        """
+        # Set up the test with a dummy authn policy and a real ACL authz
+        # policy:
+        policy = ACLAuthorizationPolicy()
+        pyramid_config.testing_securitypolicy(userid)
+        pyramid_config.set_authorization_policy(policy)
+
+        ann = factories.Annotation(shared=True,
+                                   userid='mioara',
+                                   groupid=groupid)
+        res = AnnotationContext(ann, group_service, links_service)
+
+        if permitted:
```

Comment:
This if/else could be marginally simplified to `pyramid_request.has_permission(...) is permitted`